apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommandTextBuilder.cs

```diff
@@ -255,13 +255,117 @@ public static SpannerCommandTextBuilder FromCommandText(string commandText)
+                                "or hints, or any other SQL syntax feature. If you need to use comments or hints, " +
+                                "please use DML commands. You can read more about this topic in " +
+                                "https://cloud.google.com/spanner/docs/dml-versus-mutations.");
+                        }
                         targetTable = ValidateTableName(commandSections[1], nameof(commandText));
                         // This just normalizes the command text.
                         commandText = $"{commandName} {targetTable}";
                     }
                     break;
             }
-            return new SpannerCommandTextBuilder(commandText, commandType, targetTable, extraStatements: null);
+            return new SpannerCommandTextBuilder(commandText.Trim(), commandType, targetTable, extraStatements: null);
+        }
+
+        private static string RemoveLeadingCommentsAndHints(string commandText, out bool removedCommentOrHint)
+        {
+            try
+            {
+                return commandText.Substring(IndexOfCommandStart(commandText, out removedCommentOrHint));
+            }
+            catch(Exception ex) when (ex is ArgumentOutOfRangeException || ex is IndexOutOfRangeException)
+            {
+                throw new ArgumentException($"'{commandText}' is not a recognized Spanner command.", nameof(commandText), ex);
+            }
+        }
+
+        private static int IndexOfCommandStart(string command, out bool removedCommentOrHint)
+        {
+            int currentIndex = 0;
+            int loops = 0;
+            do
+            {
+                loops++;
+                SkipWhiteSpaces();
+            }
+            while (SkipLineComment() || SkipBlockComment() || SkipHint());
+
+            removedCommentOrHint = loops > 1;
+            return currentIndex;
+
+            void SkipWhiteSpaces()
+            {
+                while (char.IsWhiteSpace(command[currentIndex]))
+                {
+                    currentIndex++;
+                }
+            }
+
+            bool SkipLineComment()
+            {
+                if (command.IndexOf('#', currentIndex, 1) >= 0 ||
+                    command.IndexOf("--", currentIndex, 2) >= 0)
+                {
+                    // TODO: If this can be escaped, let's check that it's not
+                    // or continue searching.
+                    currentIndex = command.IndexOf('\n', currentIndex);
+                    if (currentIndex >= 0) // If we didn't find it, we don't fail here.
+                    {
+                        currentIndex++;
+                    }
+                    return true;
+                }
+                return false;
+            }
+
+            bool SkipBlockComment()
+            {
+                if (command.IndexOf("/*", currentIndex, 2) >= 0 )
+                {
+                    // TODO: If this can be escaped, let's check that it's not
```

Comment:
The end of a block comment can also not be escaped, so we don't have to worry about that.